app/models/sqlalchemy/jobstore.py

```diff
@@ -0,0 +1,108 @@
+"""
+jobstore.py
+Tools for storing jobs from MolSSI SEAMM
+
+Has the SEAMM datastore class.
+"""
+
+import os
+import glob
+
+from contextlib import contextmanager
+from datetime import datetime
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker, relationship
+
+from .models import Base, Flowchart, Job, Project, JobProject
+from .util import process_flowchart, process_job
+
+
+class DataStore():
+    def __init__(self, db_path, db_type='sqlite'):
+        self.db_location = os.path.abspath(db_path)
+
+        # TODO Check if exists. If exists, check structure to see if it is correct.
+        # If exists and not correct format - error and exit.
+        if os.path.exists(self.db_location):
+            os.remove(self.db_location)
+
+        engine = create_engine(db_type + ':///' + self.db_location)
+        Base.metadata.create_all(engine)
+
+        self.Session = sessionmaker(bind=engine)
+
+    @contextmanager
+    def get_session_scope(self):
+        """Provide a transactional scope
+        Usage:
+            with self.get_session_scope() as session:
+                result = session.query(..)
+                print(result)
+            print('Query is done and committed')
+        """
+
+        session = self.Session()
+        try:
+            yield session
+            session.commit()
+        except:
+            session.rollback()
+            raise
+        finally:
+            session.close()
```

Comment:
Flask-sqlalchemy has can handle sessions for you so that you don't have to worry about it anymore. After merging this PR we can refactor with it, it will be less hassle.

Reply:
This is from the original code that I wrote before I knew about `flask-sqlalchemy`. It's not used at all with the results dashboard (only in the very beginning to build the initial datastore). This class along with some functions that are in `util.py` have some things so that jobs can be added to the datastore by specifying a directory path. I'd like to keep that in the refactor.

I was thinking to keep this so that we could interact with the database without using `flask`. Is that also possible using `flask-sqlalchemy`?

Reply:
I followed the pattern on the documentation under 'Declarative' on this page. I was trying to leave my original  sql-alchemy code relatively unchanged - https://flask.palletsprojects.com/en/1.1.x/patterns/sqlalchemy/

Reply:
It's fine, we can keep sqlachemy without flask for now. Flask-sqlalckmy is useful once we grow and we will need DB Migrations using Alembic.